x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -285,12 +325,94 @@ export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitializat
     mlLicense.fullLicenseAPIGuard(async (context, request, response) => {
       try {
         const { analyticsId } = request.params;
-        const results = await context.ml!.mlClient.callAsCurrentUser(
-          'ml.deleteDataFrameAnalytics',
-          {
-            analyticsId,
+        const { deleteTargetIndex, deleteIndexPattern } = request.query;
+        const errorsEncountered = [];
+        let destinationIndex: string | undefined;
+        let deleteTargetIndexAcknowledged: boolean = false;
+        let deleteIndexPatternAcknowledged: boolean = false;
+
+        // Check if analyticsId is valid and get destination index
+        if (deleteTargetIndex || deleteIndexPattern) {
+          try {
+            const dfa = await context.ml!.mlClient.callAsCurrentUser('ml.getDataFrameAnalytics', {
+              analyticsId,
+            });
+            if (Array.isArray(dfa.data_frame_analytics) && dfa.data_frame_analytics.length > 0) {
+              destinationIndex = dfa.data_frame_analytics[0].dest.index;
+            }
+          } catch (e) {
+            return response.customError(wrapError(e));
           }
-        );
+
+          // If user checks box to delete the destinationIndex associated with the job
+          if (destinationIndex && deleteTargetIndex) {
+            // Verify if user has privilege to delete the destination index
+            const userCanDeleteDestIndex = await userCanDeleteIndex(context, destinationIndex);
+            // If user does have privilege to delete the index, then delete the index
+            if (userCanDeleteDestIndex) {
+              try {
+                await context.ml!.mlClient.callAsCurrentUser('indices.delete', {
+                  index: destinationIndex,
+                });
+                deleteTargetIndexAcknowledged = true;
+              } catch (deleteIndexError) {
+                errorsEncountered.push({
+                  msg: i18n.translate(
```

Comment:
we don't translate error messages from endpoints

Reply:
Removed in https://github.com/elastic/kibana/pull/66934/commits/ba820793f855ac45c4109ec490bf0cfce1a98fcc